src/traversals/dfs.jl

```diff
@@ -10,7 +10,7 @@ Return `true` if graph `g` contains a cycle.
 Uses DFS.
 """
 function is_cyclic end
-@traitfn is_cyclic(g::::(!IsDirected)) = ne(g) > 0
+@traitfn is_cyclic(g::::(!IsDirected)) = (ne(g)-(nv(g)-1)) > 0
```

Comment:
I think this not enough, consider the cyclic graph of 5 vertices
```
1  -- 2       4 -- 5
  \  /
    3
```
then `nv(g) == 5` and `ne(g) == 4`, so that `ne(g) - (nv(g) - 1)) == 0`. We therefore probably need do check every connceted component separately.